### PR TITLE
Be robust against invalid utf-8 byte sequences and surrogateescape them when en- or decoding

### PR DESCRIPTION
**Pull request by pyrco**
This commit also takes the opportunity to remove Python 2 string compatibility code.

It will also remove the final left-over Python 2 compatibility in the test cases.

### CODE REVIEW

**Comment by yunzheng** [comment]
has probably nothing to do with the actual changes in the PR, but i'm not sure why pypy3.9 keep failing.

**Comment by pyrco** [comment]
> has probably nothing to do with the actual changes in the PR, but i'm not sure why pypy3.9 keep failing in the CI.

There seems to be an issue with a sub-dependency (cramjam) of the avro dependency, which fails building because the installed version of nasm and/or yasm is too old apparently.

Probably a new version got released without wheels for pyp3.9

**Comment by pyrco** [comment]
> Probably a new version got released without wheels for pypy3.9

Jup, cramjam stopped building wheels for pypy3.9

File: flow/record/adapter/sqlite.py [hunk 1]
```diff
@@ -187,7 +187,7 @@ def read_table(self, table_name: str) -> Iterator[Record]:
                         if value == 0:
                             row[idx] = None
                         elif isinstance(value, str):
-                            row[idx] = value.encode("utf-8")
+                            row[idx] = value.encode()
```
[the remark this answers is not in the document]
Comment (pyrco) [reply flow/record/adapter/sqlite.py:190]: Yes it should. I had it in there at some point, but it somehow fell-out...

**Review by yunzheng — CHANGES_REQUESTED** [review]

File: flow/record/stream.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def __init__(self, fp, flush=True):
         self.auto_flush = flush
 
     def write(self, obj):
-        buf = repr(obj).encode() + b"\n"
+        buf = repr(obj).encode(errors="surrogateescape") + b"\n"
```
[the remark this answers is not in the document]
Comment (pyrco) [reply flow/record/stream.py:43]: It's not needed indeed, `__repr__` should always return a string type, and `repr()` knows how to deal with that if there are surrogates.
I removed it.

**Review by yunzheng — APPROVED** [review]

**Pull request merged**